Sources/Client/Client/Client.swift

```diff
@@ -62,12 +71,22 @@ public final class Client: Uploader {
     ///
     /// - Parameter configuration: The configuration object with details of how the shared instance should be set up.
     public static func configureShared(_ config: Config) {
-        guard backingSharedClient == nil else {
-            ClientLogger.logAssertionFailure(
-                "`Client.shared` instance was already used. It's not possible to change its configuration."
-            )
-            return
+        if backingSharedClient != nil {
+            let errorMessage = "`Client.shared` instance was already used. "
+                + "It's not possible to change its configuration.\n"
+                + "You're either setting `Client.config` or calling `Client.configureShared` multiple times. "
+                + "You should only do this once.\n"
+                + "Break on \(#file):\(#line) to catch this issue."
+            
+            if configForSharedClient != config {
+                // Error
+                ClientLogger.logAssertionFailure(errorMessage)
```

Comment:
I don't think we should fail here.
The same example as for different environments. I think this should be in logs too.
And we should update api key with the value from the new config.
The case will be as you described, developers don't want to separate these things. A switching of environments could be possible if will update api key. Then this will work:
```swift
Client.configure(apiKey: (isDevEnvironment ? "API_KEY_DEV" : "API_KEY_STAGING"))
Client.shared.set(user: ..., token: ...)
```